core/src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -1901,14 +1901,68 @@ sepBy
 -------------------------------------------------------------------------------
 --
 -- | See 'Streamly.Internal.Data.Parser.sequence'.
---
--- /Unimplemented/
---
 {-# INLINE sequence #-}
-sequence ::
-    -- Foldable t =>
-    Fold m b c -> t (Parser m a b) -> Parser m a c
-sequence _f _p = undefined
+sequence :: MonadThrow m => Fold m b c -> D.Stream m (Parser m a b) -> Parser m a c
+sequence (Fold fstep finitial fextract) (D.Stream sstep sstate) =
+    Parser step initial extract
+    where
+    initial = do
+        fres <- finitial
+        case fres of
+            FL.Partial fs -> return $ IPartial (Nothing', sstate, fs)
+            FL.Done c -> return $ IDone c
+
+    -- state does not contain any parser
+    -- yield a new parser from the stream
+    step (Nothing', ss, fs) _ = do
```

Comment:
You can probably create a custom state for being more elaborate,

```
data SequenceState = SSInit ss fs | SSParserInit ...
```

Reply:
After the current changes, do we still need it?